controller/api/public/stat_summary.go

```diff
@@ -237,6 +264,93 @@ func (s *grpcServer) k8sResourceQuery(ctx context.Context, req *pb.StatSummaryRe
 	return resourceResult{res: &rsp, err: nil}
 }
 
+func (s *grpcServer) trafficSplitResourceQuery(ctx context.Context, req *pb.StatSummaryRequest) resourceResult {
+	requestedResource := req.GetSelector().GetResource()
+	objects, err := s.k8sAPI.GetObjects(requestedResource.Namespace, requestedResource.Type, requestedResource.Name)
+	if err != nil {
+		return resourceResult{res: nil, err: err}
+	}
+
+	var tsK8sInfo []*trafficSplitStats
+	tsMetrics := make(map[tsKey]*pb.BasicStats)
+
+	for _, object := range objects {
+		ts, ok := object.(*v1alpha1.TrafficSplit)
+		if !ok {
+			return resourceResult{res: nil, err: errors.New("Could not cast to trafficsplit")}
+		}
+
+		tsName := ts.ObjectMeta.Name
+		apex := ts.Spec.Service
+		backends := ts.Spec.Backends
+		tsStats := &trafficSplitStats{apex: apex, leaves: []leaf{}}
+		tsStats.name = tsName
+		tsStats.namespace = ts.ObjectMeta.Namespace
+
+		for _, returnedLeaf := range backends {
```

Comment:
Because getTrafficSplitMetrics doesn't use the `leaves` field of `tsStats`, it's possible to move this loop down to after the call to `getTrafficSplitMetrics`.  I think that might be helpful for simplifying this function (see next comment)